language-support/scala/bindings/src/main/scala/com/digitalasset/ledger/client/binding/binding/TemplateCompanion.scala

```diff
@@ -37,6 +37,13 @@ abstract class TemplateCompanion[T](implicit isTemplate: T <~< Template[T])
     */
   implicit final def `the TemplateCompanion`: this.type = this
 
+  /** The template's key type, or [[Nothing]] if there is no key type. */
+  type key
```

Comment:
any specific reason why it is `key` and not `Key`? Is it because it is clashing with `final case class Key...`?

Reply:
Yes, or a module `Key`.

Reply:
I see what you mean. No, the `final case class Key` explicitly written here cannot clash with `type key`. But you're allowed to have types or modules named `Key`. That's also why we have `type ContractId` in `TemplateCompanion` instead of the originally planned `type Id`: you're allowed to use `Id` as your own type name, but `ContractId` is reserved so there is no danger of collision.